### PR TITLE
USB network tethering

### PR DESCRIPTION
**Pull request by doniks**

### CODE REVIEW

File: porting/debug-build/usbnet.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  * Lastly, you need your computers IP address on this network. Following the telnet instructions this is ``192.168.2.1``.
+
+You'll need them for the following commands::
+
+   sudo sysctl net.ipv4.ip_forward=1
+   sudo iptables -t nat -A POSTROUTING -o [INTERNET] -j MASQUERADE
+   sudo iptables -A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
+   sudo iptables -A FORWARD -i [USBNETWORK] -o [INTERNET] -j ACCEPT
+
+Then, run the following command using your telnet connection on your phone::
```
Comment (bhush9) [inline porting/debug-build/usbnet.rst]: Given we are using ssh here, shouldn't this say SSH? and also we are using different ip address, no?

Comment (mikroskeem) [reply porting/debug-build/usbnet.rst]: SSH should be preferred over telnet yes. Security is important :wink:

Comment (doniks) [reply porting/debug-build/usbnet.rst]: Are we doing the ssh in the halium rootfs? I can't remember anymore. I thought that was only in the plasma mobile rootfs?! We don't have that ssh access described yet in the porting guide or do we?

I was thinking about the halium initrd. There we only have telnet. (And there I don't think security is much of a concern. )

So, maybe what we actually should do is restructure this into five separate sections:
1. observe the usb serial number debug messages
2. establish usb network 
3. establish telnet over usb network (to the initrd) 
4. establish ssh over usb (to the rootfs) 
5. establish tethering over usb network

And then reference between each other. So, e.g., 2. uses the IP's you gather from 1. and 3-4 use the network that you setup in 2

Comment (UniversalSuperBox) [reply porting/debug-build/usbnet.rst]: USB tethering should not be considered from initrd. I've added instructions to log in via SSH in #41, these instructions should be based on using that shell.

Comment (doniks) [reply porting/debug-build/usbnet.rst]: > USB tethering should not be considered from initrd. 
why?

> I've added instructions to log in via SSH in #41, these instructions should be based on using that shell.
cool

Comment (UniversalSuperBox) [reply porting/debug-build/usbnet.rst]: > why?

Initrd is pre-boot, there should never be anything you'd need internet access for in it. Once the system boots, the Telnet connection is cut off and you gain the SSH access.

You might be thinking about old UBports documenation where I included a custom initrd that exposed telnet, even once the system was started.

Comment (doniks) [reply porting/debug-build/usbnet.rst]: No, I wasn't aware of that custom initrd. What I was thinking about was simply: Here is a technology that can enable people, so lets write it down. And I have now understood that the usb tethering doesn't relate to EITHER one telnet OR ssh. So my suggestion was simply to decouple the topics if they do not need to be coupled. Here's how you do usbnet, this is how you do telnet over it, this is how you do ssh over it and that is how you do tethering.

**Comment by bhush9** [comment]
Can you fix merge-conflicts? It seems debug-build docs were changed.

**Comment by doniks** [comment]
Hey @bhush9, sorry, I know this is in need of a follow up and a clean up. I'm just not getting around to doing it. And realistically, I wont for at least a few more days.

**Pull request merged**